Marlin/src/pins/stm32f1/pins_CREALITY_V4.h

```diff
@@ -283,3 +283,26 @@
   #define NEOPIXEL_PIN                      PA13
 
 #endif
+
+// Pins for documetation and sanity checks only.
+// Changing these will not chnage the pin they are on.
+
+// Hardware UART pins
+#define UART1_TX_PIN                        PA9  // default use CH340 RX
+#define UART1_RX_PIN                        PA10 // default use CH340 TX
+#define UART2_TX_PIN                        PA2  // default use HEATER_BED_PIN
+#define UART2_RX_PIN                        PA3  // not connected
+#define UART3_TX_PIN                        PB10 // default use LCD connector
+#define UART3_RX_PIN                        PB11 // default use LCD connector
+#define UART4_TX_PIN                        PC10 // default use sdcard SDIO_D2
+#define UART4_RX_PIN                        PC11 // default use sdcard SDIO_D3
+#define UART5_TX_PIN                        PC12 // default use sdcard SDIO_CK
+#define UART5_RX_PIN                        PD2  // default use sdcard SDIO_CMD
```

Comment:
If these pins are defined based on the variant of the STM32 then we ought to define them within the HAL.

Reply:
but each build env is different not each hal....

eg we have one hal for stm32, but the various envs for stm32 based boards are not all the same. depending on the board designers IO needs...  (uart pins get moved around) 
  
The idea here is three fold

1) have a convenient place where all pin definitions are for each board  Ie the boards pin.h file
2) add these to pins debugging    (a nicety, but useful.)
3) allow for pin conflict detection.

3 being the ultimate goal, so that when a user enables an impossible combination (such as a serial port that conflicts with other io pins)  they get an immediate alert they have done something wrong. . 
vs what we have now:
The users sets an impossible combination, it doesn't work, they open up a issue and expend precious developer time looking into it to find it was a user error.